### ISSUE TITLE
Carthage stuck on updating if SSH agent isn't active

### ISSUE THREAD

**Issue by iSevenDays** [Carthage/Carthage#1705]
* `carthage version`: 
0.18.1
* `xcodebuild -version`: 
Xcode 8.2
Build version 8C38
* Are you using `--no-build`? NO
* Are you using `--no-use-binaries`? NO
* Are you using `--use-submodules`? NO

**Cartfile**
```
github "facebook/pop"
```

**Carthage Output**
```
sevens:project seven$ carthage update --verbose --platform iOS
*** Cloning pop
^C

**STUCK**
```

1. Restart MAC
2. run carthage update --verbose --platform iOS
3. See nothing happens
4. execute ssh-add -k ~/.ssh/id_rsa
5. Type password for id_rsa
6. Run carthage update --verbose --platform iOS
7. Executed successfully

**Comment by 15458434** [comment]
Same here.

**Comment by mdiep** [comment]
How can Carthage prevent this? Is there a flag we can add somewhere to not prompt?

**Comment by jliu218-ford** [comment]
yes.
https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/

**Comment by ikesyo** [comment]
Closing this due to inactivity.

**Comment by ikesyo** [comment]
Looks like we can replicate this: https://github.com/apple/swift-package-manager/pull/1839

```
env["GIT_SSH_COMMAND"] = "ssh -oBatchMode=yes"
```

**Comment by hakanyildizay** [comment]
Same here too. 
Thank you so much @iSevenDays

**Comment by ikesyo** [comment]
Addressed by #2734.
